.github/workflows/gating.yaml

```diff
@@ -101,3 +110,46 @@ jobs:
         ignore: DL3041
         failure-threshold: warning
 
+  build-image:
+    name: Build Cachi2 image and run smoketest on it
+    runs-on: ubuntu-latest
+    steps:
+      - name: add checkout action...
+        uses: actions/checkout@v3
+
+      - name: Build Cachi2 image
+        uses: redhat-actions/buildah-build@v2
+        with:
+          image: ${{ env.IMAGE_NAME }}
+          tags: ${{ github.sha }}
+          arch: amd64
+          containerfiles: |
+            ./Containerfile
+
+      - name: Check images created
+        run: buildah images | grep '${{ env.IMAGE_NAME }}'
+
+      - name: Run smoketest on built images
+        run: |
+          git clone -b ${{ env.TEST_BRANCH }} ${{ env.TEST_REPO }} ${{ env.TEST_FOLDER }}
+          podman run -t -v ${{ env.TEST_FOLDER }}:${{ env.TEST_FOLDER }} \
+            ${{ env.IMAGE_NAME }}:${{ github.sha }} \
+            fetch-deps \
+            --source ${{ env.TEST_FOLDER }} \
+            --output ./cachi2-output \
+            --package '{"path": ".", "type": "gomod"}'
+
+     # Authenticate to container image registry to push the image
+      - name: Podman Login
+        uses: redhat-actions/podman-login@v1
+        with:
+          registry: ${{ env.IMAGE_REGISTRY }}
+          username: ${{ secrets.QUAY_USER }}
+          password: ${{ secrets.QUAY_PWD }}
+
+      - name: Push image to Quay.io registry
+        run: |
+          if [[ "${{ github.ref }}" == "refs/heads/main" ]]; then
```

Comment:
This might be nicer, and useful for podman login https://stackoverflow.com/questions/58139406/only-run-job-on-specific-branch-with-github-actions

Reply:
Although I'm not sure if you can do the same thing on the "step" level :thinking:

Reply:
Looks like it works. Thanks for the tip adam!